code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingExtension.java

```diff
@@ -231,84 +211,59 @@ void handleWildcardEvents(final Event event) {
-     * The queued events are then processed in an orderly fashion.
-     * No action is taken if the provided event's value is null.
+     * Validates the passed in event and triggers additional processing based on the event type.
      *
-     * @param event The {@link Event} that needs to be queued
+     * @param eventToProcess an {@link Event} from an {@link ExtensionEventListener} to be processed
      */
-    void queueEvent(final Event event) {
-        if (event == null) {
+    void processEvent(final Event eventToProcess) {
+        if (!eventIsValid(eventToProcess)) {
+            Log.debug(LOG_TAG, SELF_TAG, "Event or EventData is null, ignoring the event.");
             return;
         }
 
-        eventQueue.add(event);
-    }
+        final Map<String, Object> configSharedState = getSharedState(MessagingConstants.SharedState.Configuration.EXTENSION_NAME, eventToProcess);
+        final Map<String, Object> edgeIdentitySharedState = getXDMSharedState(MessagingConstants.SharedState.EdgeIdentity.EXTENSION_NAME, eventToProcess);
 
-    /**
-     * Processes the queued event one by one until queue is empty.
-     *
-     * <p>
-     * Suspends processing of the events in the queue if the configuration or identity shared state is not ready.
-     * Processed events are polled out of the {@link #eventQueue}.
-     */
-    void processEvents() {
-        while (!eventQueue.isEmpty()) {
-            Event eventToProcess = eventQueue.peek();
-
-            if (eventToProcess == null) {
-                Log.debug(LOG_TAG, SELF_TAG, "Unable to process event, Event received is null.");
-                return;
-            }
-
-            final Map<String, Object> configSharedState = getSharedState(MessagingConstants.SharedState.Configuration.EXTENSION_NAME, eventToProcess);
-            final Map<String, Object> edgeIdentitySharedState = getXDMSharedState(MessagingConstants.SharedState.EdgeIdentity.EXTENSION_NAME, eventToProcess);
+        // NOTE: configuration is mandatory processing the event, so if shared state is null (pending) stop processing events
+        if (MessagingUtils.isMapNullOrEmpty(configSharedState)) {
+            Log.warning(LOG_TAG, SELF_TAG, "Could not process event, configuration shared state is pending");
+            return;
+        }
 
-            // NOTE: configuration is mandatory processing the event, so if shared state is null (pending) stop processing events
-            if (MessagingUtils.isMapNullOrEmpty(configSharedState)) {
-                Log.warning(LOG_TAG, SELF_TAG, "Could not process event, configuration shared state is pending");
-                return;
-            }
+        // NOTE: identity is mandatory processing the event, so if shared state is null (pending) stop processing events
+        if (MessagingUtils.isMapNullOrEmpty(edgeIdentitySharedState)) {
+            Log.warning(LOG_TAG, SELF_TAG, "Could not process event, identity shared state is pending");
+            return;
+        }
```

Comment:
It seems like similar checks on Configuration and EdgeIdentity state are already being done in `readyForEvent()`. Given that the event processing is 
 held (and the same event is presented again) until another event is incident when `readyForEvent()` returns false, are these two checks still needed here?

Reply:
you are correct, readyForEvent already takes care of this check. i will remove it from processEvent.